src/Utility/TextSplitter.cpp

```diff
@@ -0,0 +1,33 @@
+#include "TextSplitter.h"
+#include <iostream>
+int TextSplitter::numOfWindows(long long int textLength, int patternLength) {
+    return textLength - patternLength + 1;
+}
+
+std::pair<int, int> TextSplitter::lenOfChain(int windows) {
+    int rangeWidth = windows / numOfThreads;
+    int rangesWithBonus = windows % numOfThreads;
+    return std::make_pair(rangeWidth, rangesWithBonus - 1);
+}
+
+std::vector<std::tuple<size_t, size_t>> TextSplitter::splitText(long long int textLength, int patternLength) {
```

Comment:
Please add some docs telling what user is supposed to get within `TextSplitter::splitText`. We know it for now, but probably we'll forget it in a while.